atd-vzd/triggers/atd_txdot_units_mode_category_metadata_update.sql

```diff
@@ -0,0 +1,11 @@
+CREATE OR REPLACE FUNCTION public.atd_txdot_units_mode_category_metadata_update()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+    BEGIN
+        UPDATE atd_txdot_crashes SET atd_mode_category_metadata = get_crash_modes(crash_id)
+        WHERE (atd_txdot_crashes.crash_id = NEW.crash_id);
```

Comment:
I was reading about how `NEW` and `OLD` are available automagically. Do you know if this `NEW.crash_id` is referring to the unit's crash ID from where this is called [here](https://github.com/cityofaustin/atd-vz-data/pull/1233/files#diff-fbbc4d9a8a4ef8f9129aad0c174750571255c4d8a25a7476cebdd31d937f4278)?

I'm just curious to learn more, and I'm getting hung up on how the trigger knows what crash record to update.

Reply:
Yeah thats right, the functions called by triggers use the `NEW` and `OLD` values from the table the trigger was called on. So it would use the unit's `crash_id`. I think it works bc units share the crash id with their associated crash, so a trigger can happen on updates to the unit table and cause changes to the correct crash record by using the crash id to identify the crash record (the `WHERE` clause in the `UPDATE` statement here).

Reply:
Thanks and that's pretty rad - learning a lot in this PR! 😎